pandas/tseries/tests/test_base.py

```diff
@@ -802,6 +816,20 @@ def test_getitem(self):
             self.assert_index_equal(result, expected)
             self.assertEqual(result.freq, expected.freq)
 
+    def test_drop_duplicates_metadata(self):
+        #GH 10115
+        idx = pd.timedelta_range('1 day', '31 day', freq='D', name='idx')
+        result = idx.drop_duplicates()
+        self.assert_index_equal(idx, result)
+        self.assertEqual(idx.freq, result.freq)
+
+        idx_dup = idx.append(idx)
+        self.assertIsNone(idx_dup.freq)   # freq is reset
+        result = idx_dup.drop_duplicates()
+        self.assert_index_equal(idx, result)
+        self.assertIsNone(result.freq)
```

Comment:
I intended these tests are for  ``freq`` rather than ``name``, because ``name`` are tested in ``test_index``. Maybe ``assert_index_equal`` should check all metadata to simplify the test (I've once tried and found some tests / logics must be fixed).